src/Rules/Comparison/MatchExpressionRule.php

```diff
@@ -65,13 +66,23 @@ public function processNode(Node $node, Scope $scope): array
 					$matchCondition,
 					$armCondition->getCondition(),
 				);
-				$armConditionResult = ($this->treatPhpDocTypesAsCertain ? $armConditionScope->getType($armConditionExpr) : $armConditionScope->getNativeType($armConditionExpr));
```

Comment:
I split this in `nextArmIsDeadForType` and `nextArmIsDeadForNativeType` because
- `Match arm is unreachable` should rely on `$nextArmIsDeadForNativeType || $nextArmIsDeadForType && $this->treatPhpDocTypesAsCertain` in order to not be reported in the issue 8932.
-  `Match expression does not handle remaining` should rely only on `nextArmIsDeadForType` in order to not be reported in the issue 8937.

This way extra checks are allowed when using `treatPhpDocTypesAsCertain: false`, but phpstan are not complaining if they are missing.